src/http/RequestBody.h

```diff
@@ -56,6 +56,11 @@ class RequestBody {
 
     ~RequestBody() = default;
 
+    /// Make a new RequestBody based on `js`
+    /// TODO:UJ This would be much more efficient if this class had objJson defined as
+    ///   a const reference or pointer to const, but implementation is likely ugly.
+    RequestBody(nlohmann::json const& js) : objJson(js) {}
```

Comment:
This constructor is useless since `objJson` is populated by the existing c-tor:
```
RequestBody::RequestBody(qhttp::Request::Ptr const& req) : objJson(json::object()) {
    // This way of parsing the optional body allows requests which have no body.

    string const contentType = req->header["Content-Type"];
    string const requiredContentType = "application/json";

    if (contentType == requiredContentType) {
        string content(istreambuf_iterator<char>(req->content), {});
        if (not content.empty()) {
            try {
                objJson = json::parse(content);
                if (objJson.is_null() or objJson.is_object()) return;
            } catch (...) {
                // Not really interested in knowing specific details of the exception.
                // All what matters here is that the string can't be parsed into
                // a valid JSON object. This will be reported via another exception
                // after this block ends.
                ;
            }
            throw invalid_argument("invalid format of the request body. A simple JSON object was expected");
        }
    }
}
```
Note that not all requests have JSON body.

The second reason why the change is not welcome is because it would introduce a very big undesired effect - an extra copy (or two) of potentially very large JSON objects parsed by the POST handlers. This may happen in the Czar HTTP frontend in REST services that process inputs for the user-generated data products. Some inputs could be potentially as large as 1 GB or more.

The third reason is that a hierarchy of the `http` & `qhttp` classes has been significantly changed over the past weeks as new features were introduced to `qhttp` and an alternative HTTP server library `cpp-httplib` was added. Your change would introduce a conflict that would give you a lot of trouble during the code merge into `main` in the future.

However, I would agree with you that exposing a modifiable public data member may not be a good idea. Since this doesn't prevent your development then I will find a solution for this problem within the production branch

Reply:
I completely agree that this is horribly inefficient as is. I wanted to take advantage of your "required" functions on a subsection of an existing json message. This hack was about as minimally invasive as I could manage and keep moving forward. The TODO:UJ comment indicates that this is something that needs to be addressed, and I'm interested in seeing how you want to handle it.

Reply:
I was thinking having some sort of pointer (maybe member of the object, or parameter to "required") that could be relocated within the json "body", which should allow subsections to be examined without copying anything.